### PR TITLE
LLL: improvements (and fixes) to the lit keyword

### PR DESCRIPTION
**Pull request by axic**

### CODE REVIEW

**Comment by axic** [comment]
@zigguratt can you please base your `lit` fixes on this branch? I could also cherry-pick them into this PR once they're ready.

**Comment by axic** [comment]
#1188 will not be fixed as part of this, because it requires a lot of missing features.

**Comment by axic** [comment]
cpp-ethereum tests must be run with this version before merging.

**Comment by axic** [comment]
cpp-ethereum tests don't use LLL's `lit` keyword. I think it is safe merging this.

**Comment by zigguratt** [comment]
I think for the moment I'm the only one on the _planet_ using LLL's `lit` keyword.

File: liblll/CodeFragment.cpp [hunk 1]
```diff
@@ -281,41 +281,40 @@ void CodeFragment::constructOperation(sp::utree const& _t, CompilerState& _s)
 			bytes data;
 			for (auto const& i: _t)
 			{
-				if (ii == 1)
+				if (ii == 0)
+				{
```
Comment (pirapira) [inline liblll/CodeFragment.cpp:285]: `{` `}` are not necessary here.

File: liblll/CodeFragment.cpp [hunk 2]
```diff
@@ -281,41 +281,40 @@ void CodeFragment::constructOperation(sp::utree const& _t, CompilerState& _s)
...
+				}
+				else if (ii == 1)
 				{
 					pos = CodeFragment(i, _s);
 					if (pos.m_asm.deposit() != 1)
 						error<InvalidDeposit>();
 				}
-				else if (ii == 2 && !i.tag() && i.which() == sp::utree_type::string_type)
+				else if (!i.tag())
+				{
```
Comment (pirapira) [inline liblll/CodeFragment.cpp:296]: here as well.

File: liblll/CodeFragment.cpp (hunk 1 above)
Comment (axic) [reply liblll/CodeFragment.cpp:285]: Yes, but I think it makes a bit more readable.

File: liblll/CodeFragment.cpp [hunk 3]
```diff
@@ -281,41 +281,40 @@ void CodeFragment::constructOperation(sp::utree const& _t, CompilerState& _s)
...
 					{
-						data.resize(data.size() + 32);
-						*(h256*)(&data.back() - 31) = (u256)bi;
+						bytes tmp = toCompactBigEndian(bi);
+						data.insert(data.end(), tmp.begin(), tmp.end());
 					}
 				}
-				else if (ii)
+				else
+				{
```
Comment (pirapira) [inline liblll/CodeFragment.cpp:316]: and here too.

File: liblll/CodeFragment.cpp (hunk 1 above)
Comment (pirapira) [reply liblll/CodeFragment.cpp:285]: Wait, maybe `ii++` is missing?

Comment (axic) [reply liblll/CodeFragment.cpp:285]: Fixed.

**Review by pirapira — APPROVED** [review]
Looks good to me.

**Comment by axic** [comment]
@zigguratt can you please double check again this PR against all your LLL source codes?

**Comment by zigguratt** [comment]
@axic I'm out for the day today. I'll have to check tomorrow morning.

**Comment by zigguratt** [comment]
All of my current LLL source compiles on this branch. So I'd say go ahead and merge.

**Pull request merged**